sdk/python/kfp/v2/components/experimental/pipeline_channel.py

```diff
@@ -0,0 +1,321 @@
+    """
+    operator: str
+    operand1: Union['PipelineParameterChannel', type_utils.PARAMETER_TYPES]
+    operand2: Union['PipelineParameterChannel', type_utils.PARAMETER_TYPES]
+
+
+class PipelineChannel:
+    """Represents a future value that is passed between pipeline components.
+
+    A PipelineChannel object can be used as a pipeline function argument so that
+    it will be a pipeline artifact or parameter that shows up in ML Pipelines
+    system UI. It can also represent an intermediate value passed between
+    components.
+
+    Attributes:
+      name: The name of the pipeline channel.
+      channel_type: The type of the pipeline channel.
+      op_name: The name of the operation that produces the pipeline channel.
+        None means it is not produced by any operator, so if None, either user
+        constructs it directly (for providing an immediate value), or it is a
+        pipeline function argument.
+      pattern: The serialized string regex pattern this pipeline channel created
+        from.
+    """
+
+    def __init__(
+        self,
+        name: str,
+        channel_type: Union[str, Dict],
+        op_name: Optional[str] = None,
+    ):
+        """Inits a PipelineChannel instance.
+
+        Args:
+          name: The name of the pipeline channel.
+          channel_type: The type of the pipeline channel.
+          op_name: Optional; The name of the operation that produces the
+            pipeline channel.
+
+        Raises:
+          ValueError: If name or op_name contains invalid characters.
+          ValueError: If both op_name and value are set.
+        """
+        valid_name_regex = r'^[A-Za-z][A-Za-z0-9\s_-]*$'
+        if not re.match(valid_name_regex, name):
+            raise ValueError(
+                'Only letters, numbers, spaces, "_", and "-" are allowed in the '
+                'name. Must begin with a letter. Got name: {}'.format(name)
+            )
+
+        self.name = name
+        self.channel_type = channel_type
+        # ensure value is None even if empty string or empty list/dict
+        # so that serialization and unserialization remain consistent
+        # (i.e. None => '' => None)
+        self.op_name = op_name or None
+
+    @property
+    def full_name(self) -> str:
+        """Unique name for the PipelineChannel."""
+        return f'{self.op_name}-{self.name}' if self.op_name else self.name
+
+    @property
+    def pattern(self) -> str:
+        """Unique pattern for the PipelineChannel."""
+        return str(self)
+
+    def __str__(self) -> str:
+        """String representation of the PipelineChannel.
+
+        The string representation is a string identifier so we can mix
+        the PipelineChannel inline with other strings such as arguments.
+        For example, we can support: ['echo %s' % param] as the
+        container command and later a compiler can replace the
+        placeholder '{{pipeline_channel:op=%s;name=%s;type=%s}}' with
```

Comment:
For my education - I assume compiler does this so we can support things like concatenation of params and strings?